### PR TITLE
Migrate JS Package

### PR DESCRIPTION
**Pull request by wanlwanl**
Fix: https://github.com/Azure/azure-sdk-tools/issues/9510

### CODE REVIEW

**Comment by wanlwanl** [comment]
Tested locally.
found 2 issues in migration tools.
https://github.com/Azure/azure-sdk-for-js/issues/32202
https://github.com/Azure/azure-sdk-for-js/issues/32200

@kazrael2119  Please help to test. in windows platform, you need to modify it as a workaround: https://github.com/Azure/azure-sdk-for-js/pull/32201/files#diff-b5c6091a6b6c3d41963de6bcb4443000186b55d3c8f76469ed31037cb29134fd

File: tools/js-sdk-release-tools/src/common/migration.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { logger } from "../utils/logger";
+import { runCommand, runCommandOptions } from "./utils";
+import { getNpmPackageInfo } from "./npmUtils";
+import { ensureDir } from "fs-extra";
+import { posix } from "path";
+
+// TODO: remove when emitter is ready
+export async function migratePackage(packageDirectory: string): Promise<void> {
```
Comment (qiaozha) [inline tools/js-sdk-release-tools/src/common/migration.ts:8]: should we detect if the module is already esm?

**Comment by qiaozha** [comment]
@mpodwysocki I wonder if we should wait for the migration command for samples update to be ready?

File: tools/js-sdk-release-tools/src/common/migration.ts (hunk 1 above)
Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/common/migration.ts:8]: e.g. https://github.com/Azure/azure-sdk-for-js/pull/32201/commits/33f43d4e0592e41c3ba1667bcd2210a7ce224f03

Comment (wanlwanl) [reply tools/js-sdk-release-tools/src/common/migration.ts:8]: and they are idemptent

**Comment by wanlwanl** [comment]
@qiaozha please note that there're 2 issues may need to fix by https://github.com/Azure/azure-sdk-tools/pull/9511#issuecomment-2540632570. but I added a workaround, they won't impact us for now.

File: tools/js-sdk-release-tools/src/common/migration.ts (hunk 1 above)
Comment (qiaozha) [reply tools/js-sdk-release-tools/src/common/migration.ts:8]: Got it, thanks for the info and pointers.

**Review by qiaozha — APPROVED** [review]

**Comment by wanlwanl** [comment]
@weshaggard could you help? I got the same issue twices recently

**Comment by wanlwanl** [comment]
@chidozieononiwu, @mikeharder: could you please help me?

**Comment by weshaggard** [comment]
I've re-enabld the ci pipeline so it should trigger for future PRs

**Pull request merged**